webapp/graphite/render/datalib.py

```diff
@@ -109,64 +109,24 @@ def getInfo(self):
 
 @logtime()
 def _fetchData(pathExpr, startTime, endTime, now, requestContext, seriesList):
-  if settings.REMOTE_PREFETCH_DATA:
-    matching_nodes = [node for node in STORE.find(pathExpr, startTime, endTime, local=True)]
-
-    # inflight_requests is only present if at least one remote store
-    # has been queried
-    if 'inflight_requests' in requestContext:
-      fetches = requestContext['inflight_requests']
-    else:
-      fetches = {}
-
-    def result_queue_generator():
-      for node in matching_nodes:
-        if node.is_leaf:
-          yield (node.path, node.fetch(startTime, endTime, now, requestContext))
-
-      log.info(
-        'render.datalib.fetchData:: result_queue_generator got {count} fetches'
-        .format(count=len(fetches)),
-      )
-      for key, fetch in fetches.iteritems():
-        log.info(
-          'render.datalib.fetchData:: getting results of {host}'
-          .format(host=key),
-        )
-
-        if isinstance(fetch, FetchInProgress):
-          fetch = fetch.waitForResults()
-
-        if fetch is None:
-          log.info('render.datalib.fetchData:: fetch is None')
-          continue
-
-        for result in fetch:
-          if result['pathExpression'] == pathExpr:
-            yield (
-              result['path'],
-              (
-                (result['start'], result['end'], result['step']),
-                result['values'],
-              ),
-            )
-
-    result_queue = result_queue_generator()
-  else:
-    matching_nodes = [node for node in STORE.find(pathExpr, startTime, endTime, local=requestContext['localOnly'])]
-    result_queue = [
-      (node.path, node.fetch(startTime, endTime, now, requestContext))
-      for node in matching_nodes
-      if node.is_leaf
-    ]
-
-  log.info("render.datalib.fetchData :: starting to merge")
+  matching_nodes = STORE.find(
+    pathExpr, startTime, endTime,
+    local=requestContext['localOnly'],
```

Comment:
This seems like it's going to trigger find requests to all the remote hosts.  Avoiding those extra requests when using prefetch was part of the optimization we did to improve performance in 1.x